src/HttpClientConfigurator.php

```diff
@@ -98,4 +101,28 @@ public function setApiKey(string $apiKey)
 
         return $this;
     }
+
+    /**
+     * @param Plugin $plugin
+     *
+     * @return HttpClientConfigurator
+     */
+    public function appendPlugin(Plugin $plugin)
```

Comment:
Since we are PHP 7, we could use variadic here. Not sure if it's better though, but might make the called methods shorter. WDYT?